src/game_events/handlers.cpp

```diff
@@ -241,10 +241,31 @@ struct filter_attack : public event_filter {
 	{
 		const unit_map& units = resources::gameboard->units();
 		const auto& loc = first_ ? event_info.loc1 : event_info.loc2;
-		auto unit = units.find(loc);
-		if(unit != units.end() && loc.matches_unit(unit)) {
-			const config& attack = event_info.data.child(first_ ? "first" : "second");
-			return swf_.empty() || matches_special_filter(attack, swf_);
+		const auto& loc_d = first_ ? event_info.loc2 : event_info.loc1;
+		auto unit_a = units.find(loc);
+		assert(unit_a.valid());
```

Comment:
Unless you can explain why it will always find a unit, replacing `if(unit != units.end ...)` with an `assert` isn't going to be approved.

Even with an explanation, the defensive code is preferable.

Reply:
line 251 if(unit_a != units.end() && loc.matches_unit(unit_a)) { is here, i not replace but added assert but i can remove that.

Reply:
assert removed